nomos-core/chain-defs/src/block/mod.rs

```
@@ -1,30 +1,51 @@
+use core::fmt::Debug;
+
 use ::serde::{Deserialize, Serialize, de::DeserializeOwned};
 use bytes::Bytes;
-use poseidon2::Fr;
+use cryptarchia_engine::Slot;
+use ed25519_dalek::Verifier as _;
+
+use crate::{
+    codec::SerdeOp,
+    header::{ContentId, Header, HeaderId},
+    mantle::{Transaction, TxHash},
+    proofs::leader_proof::{Groth16LeaderProof, LeaderProof as _},
+    utils::merkle,
+};
+
+const MAX_TRANSACTIONS: usize = 1024;
 
-use crate::{codec::SerdeOp, header::Header};
-pub type TxHash = [u8; 32];
 pub type BlockNumber = u64;
 pub type SessionNumber = u64;
 
-/// A block proposal
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Failed to serialize: {0}")]
+    Serialisation(#[from] crate::codec::Error),
+    #[error("Signature error: {0}")]
+    Signature(String),
```

better to bubble up the underlying error